go/pkg/client/cas.go

```diff
@@ -99,9 +168,27 @@ func (c *Client) UploadIfMissing(ctx context.Context, data ...*chunker.Chunker)
 		})
 	}
 
-	log.V(2).Info("Waiting for remaining jobs")
+	log.V(2).Info("Waiting for remaining new uploads")
 	err = eg.Wait()
 	log.V(2).Info("Done")
+	log.V(2).Infof("Waiting for remaining %v previous uploads", len(previousUploads))
+	for _, dg := range previousUploads {
+		// Wait on all uploads, but return the first error.
+		mu := locks[dg]
+		mu.RLock()
+		defer mu.RUnlock()
+		if err != nil {
+			continue
+		}
+		if v, _ := c.casUploadErrors.Load(dg); v != nil {
+			e, ok := v.(error)
+			if !ok {
+				err = fmt.Errorf("unexpected type found in error map")
+			}
+			err = e
+		}
+	}
+	log.V(2).Info("Done")
```

Comment:
Can you make this more descriptive about what was finished?

Reply:
Yep!